### PR TITLE
Visual studio warnings fix

### PR DESCRIPTION
**Pull request by anton-dirac**
This patch fixes and removes warnings when compiling google-benchmark with Visual Studio.
The first commit changes the warning level from Wall to W4 when MSVC is used. Wall on MSVC is really strict and gives 10000+ warnings (most of them in windows and STL headers...). 

The largest change is to the OS macros to avoid a new clash in Shlwapi.h, see babe4e9.
While fixing the warnings I also found a bug, see: 3c16f40.

### CODE REVIEW

File: src/benchmark.cc [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #include "benchmark/benchmark.h"
 #include "internal_macros.h"
 
-#ifndef OS_WINDOWS
+#ifndef BENCHMARK_OS_WINDOWS
```
Comment (dominichamon) [inline src/benchmark.cc]: this is a nice cleanup. we should do the others to match.

hint.

File: src/benchmark.cc [hunk 2]
```diff
@@ -623,7 +623,7 @@ void RunBenchmark(const benchmark::internal::Benchmark::Instance& b,
 
   for (int i = 0; i < FLAGS_benchmark_repetitions; i++) {
     std::string mem;
-    while (true) {
+    for (;;) {
```
Comment (dominichamon) [inline src/benchmark.cc]: this gives a warning in msvc? oy vey.

**Comment by dominichamon** [comment]
change looks good. when the CLA is signed i'll pull it in.

thanks!

File: src/benchmark.cc (hunk 2 above)
Comment (anton-dirac) [inline src/benchmark.cc]: Yes, with W4 it does... (warning C4127: conditional expression is constant)
An alternative would be to disable this warning in the CMake file.

**Comment by anton-dirac** [comment]
I signed it! 
(Corporation DiracResearch)

**Pull request merged**